### PR TITLE
Delete/ingestor

### PR DESCRIPTION
**Pull request by Koustavd18**

### CODE REVIEW

File: src/pages/Systems/MachineInfo.tsx [hunk 1]
```diff
@@ -83,10 +85,17 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
 };
 
 const IngestorInfo = () => {
+	const [opened, setOpened] = useState(false);
 	const [recentRecord] = useClusterStore((store) => store.currentMachineRecentRecord);
 	const [ingestorMachines] = useClusterStore((store) => store.ingestorMachines);
 	const ingestor = _.find(ingestorMachines, (ingestor) => ingestor.domain_name === recentRecord?.address);
 	const error = ingestor ? ingestor.error : null || null;
+	const deleteUrl = recentRecord?.address as string;
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:93]: You shouldn't use 'as' like this which is falsy. Fix the ts errors. Straight forward.

File: src/pages/Systems/styles/Systems.module.css [hunk 2]
```diff
@@ -1,253 +1,262 @@
...
+	flex-direction: row;
+	justify-content: space-between;
+	width: 100%;
+	padding: 1rem 1rem 0rem 0.5rem;
 }
 
 .sectionTitle {
-    font-size: 1.2rem;
-    font-weight: 600;
+	font-size: 1.2rem;
```
Comment (balaji-jr) [inline src/pages/Systems/styles/Systems.module.css:18]: Please configure your formatting settings and let's make sure we don't commit unwanted changes.

File: src/pages/Systems/styles/Systems.module.css [hunk 3]
```diff
@@ -1,253 +1,262 @@
...
-    color: var(--mantine-color-gray-6);
-}
\ No newline at end of file
+	font-size: var(--mantine-font-size-md);
+	color: var(--mantine-color-gray-6);
+}
+
+.deleteIcon {
+	border: 1px solid gray;
+	padding: 1.5px;
```
Comment (balaji-jr) [inline src/pages/Systems/styles/Systems.module.css:256]: Use rem.

File: src/pages/Systems/MachineInfo.tsx [hunk 4]
```diff
@@ -98,16 +107,24 @@ const IngestorInfo = () => {
...
+					<Stack style={{ width: '85%' }}>
+						<Group justify="flex-end">
+							<Button onClick={() => setOpened(true)} className={classes.deleteIcon} color="white">
+								<IconTrash height="18" stroke={1.5} />
+							</Button>
+						</Group>
+					</Stack>
+				) : null}
 			</Stack>
+			<IngestorDeleteModal modalOpened={opened} closeModal={closeModal} IngestorAddress={deleteUrl} />
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:120]: Use proper naming conventions.

File: src/pages/Systems/MachineInfo.tsx [hunk 5]
```diff
@@ -98,16 +107,24 @@ const IngestorInfo = () => {
 						</ThemeIcon>
 					</Tooltip>
 				)}
+				{!ingestor?.reachable ? (
+					<Stack style={{ width: '85%' }}>
+						<Group justify="flex-end">
+							<Button onClick={() => setOpened(true)} className={classes.deleteIcon} color="white">
+								<IconTrash height="18" stroke={1.5} />
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:114]: use rem for sizes

File: src/pages/Systems/MachineInfo.tsx [hunk 6]
```diff
@@ -98,16 +107,24 @@ const IngestorInfo = () => {
 						</ThemeIcon>
 					</Tooltip>
 				)}
+				{!ingestor?.reachable ? (
+					<Stack style={{ width: '85%' }}>
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:111]: Use flex. Why the fixed width percentage ?

File: src/pages/Systems/MachineInfo.tsx [hunk 7]
```diff
@@ -83,10 +85,17 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
...
 
 const IngestorInfo = () => {
+	const [opened, setOpened] = useState(false);
 	const [recentRecord] = useClusterStore((store) => store.currentMachineRecentRecord);
 	const [ingestorMachines] = useClusterStore((store) => store.ingestorMachines);
 	const ingestor = _.find(ingestorMachines, (ingestor) => ingestor.domain_name === recentRecord?.address);
 	const error = ingestor ? ingestor.error : null || null;
+	const deleteUrl = recentRecord?.address as string;
+
+	const closeModal = () => {
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:95]: `useCallback` here

File: src/pages/Systems/MachineInfo.tsx [hunk 8]
```diff
@@ -83,10 +85,17 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
 };
 
 const IngestorInfo = () => {
+	const [opened, setOpened] = useState(false);
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:88]: Use proper variable names

File: src/pages/Systems/IngestorDeleteModal.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export default function IngestorDeleteModal(props: {
+	IngestorAddress: string;
+	modalOpened: boolean;
+	closeModal: () => void;
+}) {
+	const { deleteIngestorMutation, deleteIngestorIsLoading } = useDeleteIngestor();
+
+	const deleteFn = () => {
```
Comment (balaji-jr) [inline src/pages/Systems/IngestorDeleteModal.tsx:30]: `useCallback` here

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Systems/MachineInfo.tsx (hunk 5 above)
Comment (Koustavd18) [reply src/pages/Systems/MachineInfo.tsx:114]: <IconTrash> height prop doesnt allow css units it takes only number as input

Comment (balaji-jr) [reply src/pages/Systems/MachineInfo.tsx:114]: use size like everywhere else.

**Comment by balaji-jr** [comment]
I am still seeing the indentation changes.

File: src/pages/Systems/MachineInfo.tsx [hunk 10]
```diff
@@ -83,31 +84,50 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
...
+						)}
+					</Group>
+					{!ingestor?.reachable ? (
+						<Button onClick={openModal} className={classes.deleteIcon} color="white">
+							<IconTrash size={'1rem'} stroke={1.5} />
+						</Button>
+					) : null}
+				</Group>
 			</Stack>
+			<DeleteIngestorModal modalOpened={openDeleteModal} closeModal={closeModal} IngestorAddress={deleteUrl} />
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:123]: Still not using the correct case.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Systems/MachineInfo.tsx [hunk 11]
```diff
@@ -83,31 +84,50 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
...
+	const [openDeleteModal, setOpenDeleteModal] = useState(false);
 	const [recentRecord] = useClusterStore((store) => store.currentMachineRecentRecord);
 	const [ingestorMachines] = useClusterStore((store) => store.ingestorMachines);
 	const ingestor = _.find(ingestorMachines, (ingestor) => ingestor.domain_name === recentRecord?.address);
 	const error = ingestor ? ingestor.error : null || null;
+	const deleteUrl = recentRecord?.address ?? '';
+
+	const closeModal = useCallback(() => {
+		setOpenDeleteModal(false);
+	}, [openDeleteModal]);
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:96]: why openDeleteModal as a dependency here ?
I would suggest that you go through the useCallback documentation.

File: src/pages/Systems/MachineInfo.tsx [hunk 12]
```diff
@@ -83,31 +84,50 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
...
+
+	const openModal = useCallback(() => {
+		setOpenDeleteModal(true);
+	}, [openDeleteModal]);
+
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:101]: How about toggleModal that handles both opening and closing modal.

File: src/pages/Systems/MachineInfo.tsx [hunk 13]
```diff
@@ -83,31 +84,50 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
 };
 
 const IngestorInfo = () => {
+	const [openDeleteModal, setOpenDeleteModal] = useState(false);
 	const [recentRecord] = useClusterStore((store) => store.currentMachineRecentRecord);
 	const [ingestorMachines] = useClusterStore((store) => store.ingestorMachines);
 	const ingestor = _.find(ingestorMachines, (ingestor) => ingestor.domain_name === recentRecord?.address);
 	const error = ingestor ? ingestor.error : null || null;
+	const deleteUrl = recentRecord?.address ?? '';
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:92]: use more appropriate variable name.

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+	if (url.endsWith('/')) {
+		url = url.slice(0, -1);
+	}
+
+	return url;
+}
+
+const ModalTitle = () => {
+	return <Text style={{ fontWeight: 600, marginLeft: '0.5rem' }}>Confirm Delete</Text>;
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:21]: I don't think the indentation spaces used here are correct. 
Please follow the existing configuration.

File: src/pages/Systems/MachineInfo.tsx (hunk 13 above)
Comment (Koustavd18) [reply src/pages/Systems/MachineInfo.tsx:92]: recentRecord is already returning the currentMachineRecentRecord.

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+		url = url.slice(0, -1);
+	}
+
+	return url;
+}
+
+const { setCurrentMachine } = clusterStoreReducers;
+
+const ModalTitle = () => {
+	return <Text style={{ fontWeight: 600, marginLeft: '0.4.25rem' }}>Confirm Delete</Text>;
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:24]: marginLeft value is not a correct string.

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+export default function DeleteIngestorModal(props: {
+	ingestorAddress: string | null;
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:28]: Get this from the store from currentMachine.domain_name

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+	modalOpened: boolean;
+	closeModal: () => void;
+}) {
+	const { deleteIngestorMutation, deleteIngestorIsLoading } = useDeleteIngestor();
+	const { getClusterInfoRefetch } = useClusterInfo();
+	const [, setClusterStore] = useClusterStore((store) => store.currentMachine);
+
+	const deleteIngestorSuccess = () => {
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:36]: useCallback here

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+		props.closeModal();
+		setClusterStore((store) => setCurrentMachine(store, store.querierMachine.domain_name, 'querier'));
+		getClusterInfoRefetch();
+	};
+
+	const deleteIngestor = useCallback(() => {
+		if (!props.ingestorAddress) return;
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:43]: empty line after exit condition.

File: src/pages/Systems/DeleteIngestorModal.tsx (hunk 17 above)
Comment (balaji-jr) [reply src/pages/Systems/DeleteIngestorModal.tsx:36]: name should be onDeleteIngestorSuccess

File: src/pages/Systems/MachineInfo.tsx [hunk 19]
```diff
@@ -83,31 +84,50 @@ const InfoItem = (props: { title: string; value: string; width?: string; loading
 };
 
 const IngestorInfo = () => {
+	const [openDeleteModal, setOpenDeleteModal] = useState(false);
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:87]: appropriate variable name here. 
openDeleteModal sounds like a function here.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+
+const ModalTitle = () => {
+	return <Text style={{ fontWeight: 600, marginLeft: '0 4.25rem' }}>Confirm Delete</Text>;
+};
+
+export default function DeleteIngestorModal(props: { modalOpened: boolean; closeModal: () => void }) {
+	const { deleteIngestorMutation, deleteIngestorIsLoading } = useDeleteIngestor();
+	const { getClusterInfoRefetch } = useClusterInfo();
+	const [, setClusterStore] = useClusterStore((store) => store.currentMachine);
+	const [ingestorAddress] = useClusterStore((store) => store.currentMachine);
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:31]: rmv the repetitions

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+
+	const onDeleteIngestorSuccess = useCallback(() => {
+		props.closeModal();
+		setClusterStore((store) => setCurrentMachine(store, store.querierMachine.domain_name, 'querier'));
+		getClusterInfoRefetch();
+	}, [ingestorAddress]);
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:37]: why the dependency here ?

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+
+	const deleteIngestor = useCallback(() => {
+		if (!ingestorAddress) return;
+
+		deleteIngestorMutation({
+			ingestorUrl: sanitizeIngestorUrl(ingestorAddress),
+			onSuccess: onDeleteIngestorSuccess,
+		});
+	}, [ingestorAddress]);
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:46]: why not onDeleteIngestorSuccess as a dependency here ?

File: src/pages/Systems/DeleteIngestorModal.tsx [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+		url = url.slice(0, -1);
+	}
+
+	return url;
+}
+
+const { setCurrentMachine } = clusterStoreReducers;
+
+const ModalTitle = () => {
+	return <Text style={{ fontWeight: 600, marginLeft: '0 4.25rem' }}>Confirm Delete</Text>;
```
Comment (balaji-jr) [inline src/pages/Systems/DeleteIngestorModal.tsx:24]: Still the value is not correct.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

**Review by balaji-jr — DISMISSED** [review]

**Review by nikhilsinhaparseable — DISMISSED** [review]
LGTM

File: src/pages/Systems/MachineInfo.tsx [hunk 24]
```diff
@@ -108,7 +108,7 @@ const IngestorInfo = () => {
 							</Tooltip>
 						)}
 					</Group>
-					{!ingestor?.reachable ? (
+					{ingestor && !ingestor?.reachable ? (
```
Comment (balaji-jr) [inline src/pages/Systems/MachineInfo.tsx:111]: This doesn't fix the issue ie: showing empty data on machine info.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

**Review by balaji-jr — DISMISSED** [review]

**Pull request merged**